generator/bench/Bench/Pos/Criterion/Block/Logic.hs

```diff
@@ -0,0 +1,244 @@
+module Bench.Pos.Criterion.Block.Logic
+    ( runBenchmark
+    ) where
+
+import           Universum
+
+import           Control.Monad.Random.Strict (evalRandT, mapRandT)
+import           Criterion.Main (Benchmark, Benchmarkable, bench, bgroup,
+                     defaultConfig, defaultMainWith, env, nf, nfIO)
+import           Criterion.Types (Config (..), Verbosity (..))
+import qualified Data.List.NonEmpty as NE
+import           Data.Time.Units (convertUnit)
+import           Serokell.Util.Verify (isVerSuccess)
+import           System.Random (newStdGen)
+import           System.Wlog (LoggerName (..))
+
+import           Mockable.CurrentTime (realTime)
+
+import           Pos.AllSecrets (mkAllSecretsSimple)
+import           Pos.Block.Logic.Integrity (VerifyBlockParams (..),
+                     VerifyHeaderParams (..), verifyBlock, verifyHeader)
+import           Pos.Block.Logic.VAR (VerifyBlocksContext,
+                     getVerifyBlocksContext', rollbackBlocks,
+                     verifyAndApplyBlocks, verifyBlocksPrefix)
+import           Pos.Core (Block, EpochOrSlot (..), SlotId, getBlockHeader)
+import           Pos.Core.Chrono (NE, OldestFirst (..), nonEmptyNewestFirst)
+import           Pos.Core.Common (BlockCount (..), unsafeCoinPortionFromDouble)
+import           Pos.Core.Configuration (genesisBlockVersionData, genesisData,
+                     genesisSecretKeys, slotSecurityParam)
+import           Pos.Core.Genesis (FakeAvvmOptions (..), GenesisData (..),
+                     GenesisInitializer (..), TestnetBalanceOptions (..))
+import           Pos.Core.Slotting (Timestamp (..), epochIndexL, getEpochOrSlot)
+import           Pos.Core.Update (BlockVersionData (..))
+import           Pos.Crypto.Configuration (ProtocolMagic (..))
+import           Pos.DB (getTipHeader)
+import           Pos.DB.DB (initNodeDBs)
+import           Pos.Generator.Block (BlockGenParams (..), TxGenParams (..),
+                     genBlockNoApply, genBlocks, mkBlockGenContext)
+import           Pos.Launcher.Configuration (ConfigurationOptions (..),
+                     HasConfigurations, defaultConfigurationOptions,
+                     withConfigurationsM)
+import           Pos.Lrc.Context (lrcActionOnEpochReason)
+import qualified Pos.Lrc.DB as LrcDB
+import           Pos.Txp.Logic.Global (txpGlobalSettings)
+import           Pos.Util.CompileInfo (withCompileInfo)
+
+import           Test.Pos.Block.Logic.Emulation (runEmulation, sudoLiftIO)
+import           Test.Pos.Block.Logic.Mode (BlockTestContext, BlockTestMode,
+                     TestParams (..), initBlockTestContext, runBlockTestMode)
+import           Test.Pos.Block.Logic.Util (satisfySlotCheck)
+
+-- | Criterion configuration
+config :: Config
+config = defaultConfig
+    { reportFile = Just "verification.html"
+    , resamples = 10
+    , timeLimit = 5.0
+    , verbosity = Verbose
+    }
```

Comment:
Seems bad to write this here. Should be given as command-line args no?

Reply:
ok